yearn/contracts/CurveVoterProxyStrategy.sol

```diff
@@ -0,0 +1,511 @@
+    ///         last time its core position(s) were adjusted. Examples include
+    ///         unwrapping extra rewards. This call is only used during normal
+    ///         operation of a strategy, and should be optimized to minimize
+    ///         losses as much as possible. This strategy implements the
+    ///         aforementioned behavior by harvesting the Curve pool's gauge
+    ///         CRV rewards, using most of the CRV tokens to buy one of the
+    ///         Curve pool's accepted token, and depositing that token back to
+    ///         the Curve pool. This way the strategy is gaining new vault's
+    ///         underlying tokens thus making the profit. A small portion of
+    ///         CRV rewards (defined by keepCRV param) is transferred to the
+    ///         voter contract to increase CRV boost and get more CRV rewards
+    ///         in the future. Apart from that, this strategy also sells
+    ///         the Curve pool's gauge additional rewards, generated using the
+    ///         Synthetix staking rewards contract.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    /// @return profit Amount of realized profit.
+    /// @return loss Amount of realized loss.
+    /// @return debtPayment Amount of repaid debt. The value of debtPayment
+    ///         should be less than or equal to debtOutstanding. It is okay for
+    ///         it to be less than debtOutstanding, as that should only used as
+    ///         a guide for how much is left to pay back. Payments should be
+    ///         made to minimize loss from slippage, debt, withdrawal fees, etc.
+    function prepareReturn(uint256 debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 profit,
+            uint256 loss,
+            uint256 debtPayment
+        )
+    {
+        // Get the initial balance of the vault's underlying token under
+        // strategy management.
+        uint256 initialWantBalance = want.balanceOf(address(this));
+
+        // Harvest the CRV rewards from the Curve pool's gauge.
+        IStrategyProxy(strategyProxy).harvest(tbtcCurvePoolGauge);
+
+        // Buy WBTC using harvested CRV tokens.
+        uint256 crvBalance = IERC20(crv).balanceOf(address(this));
+        if (crvBalance > 0) {
+            // Deposit a portion of CRV to the voter to gain CRV boost.
+            crvBalance = adjustCRV(crvBalance);
+
+            IERC20(crv).safeApprove(dex, 0);
+            IERC20(crv).safeApprove(dex, crvBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = crv;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(dex).swapExactTokensForTokens(
```

Comment:
I think we can pass just 0 instead of `uint256(0)`. The last param is `deadline` and the docs says "Unix timestamp after which the transaction will revert.". If the transaction is happening now, won't it be reverted immediately?

Reply:
See https://github.com/keep-network/tbtc-v2/pull/8/commits/897d8bcddabf6791d3ccab496ee4a68024ac0c1b. However I think we are talking about `minAmount` parameter, not the `deadline`. The `deadline` is indeed the last one but we pass `now` here. `now` means that the swap is valid only within the block in which it was created.

Reply:
I mentioned two points: one for `uint256(0)` which you resolved and another one was for `deadline`, which you answered. Thx, looks good!